### PR TITLE
[TPC-H] Dask query 14

### PR DESCRIPTION
**Pull request by milesgranger**
Part of https://github.com/coiled/dask-engineering/issues/234

This works for both dask.dataframe and dask-expr.

### CODE REVIEW

File: tests/tpch/test_dask.py [hunk 1]
```diff
@@ -306,3 +306,49 @@ def test_query_7(client, dataset_path, fs):
...
+    where
+        l_partkey = p_partkey
+        and l_shipdate >= date '1995-09-01'
+        and l_shipdate < date '1995-09-01' + interval '1' month
+    """
+    lineitem = dd.read_parquet(dataset_path + "lineitem", filesystem=fs)
+    part = dd.read_parquet(dataset_path + "part", filesystem=fs)
+
+    shipdate_from = datetime.strptime("1995-09-01", "%Y-%m-%d")
+    shipdate_to = shipdate_from + timedelta(days=365 / 12)
```
Comment (hendrikmakait) [inline tests/tpch/test_dask.py:332]:
```suggestion
    shipdate_to = datetime.strptime("1995-10-01", "%Y-%m-%d")
```

File: tests/tpch/test_dask.py [hunk 2]
```diff
@@ -306,3 +306,49 @@ def test_query_7(client, dataset_path, fs):
...
+
+    # Promo revenue by line; CASE clause
+    table["promo_revenue"] = 0.00
+    mask = table.p_type.str.match("PROMO*")
+    table["promo_revenue"] = table.promo_revenue.where(
+        mask, table.l_extendedprice * (1 - table.l_discount)
+    )
+
+    # aggregate promo revenue calculation
+    _ = round(
```
Comment (hendrikmakait) [inline tests/tpch/test_dask.py:347]: nit: Can we move the `round` into the computation?

**Review by hendrikmakait — CHANGES_REQUESTED** [review]
One fix-up needed for datetime arithmetics.

File: tests/tpch/test_dask.py (hunk 2 above)
Comment (milesgranger) [reply tests/tpch/test_dask.py]: the `dd.Scalar<...>` has doesn't have .round method, but could do with using to_delayed the computing. :+1:

**Review by hendrikmakait — APPROVED** [review]
Thanks, @milesgranger!

**Pull request merged**